### PR TITLE
ordNub for extra stanzas

### PR DESCRIPTION
**Pull request by gbaz**
Resolves https://github.com/haskell/cabal/issues/8458

Also moves the other uses of set-based nubbing in that file to ordNub, which should hopefully be a strict improvement.

* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] Tests

### CODE REVIEW

File: Cabal/src/Distribution/Simple/Build.hs [hunk 1]
```diff
@@ -434,52 +433,36 @@ generateCode codeGens nm pdesc bi lbi clbi verbosity = do
...
-        old = Set.fromList $ otherModules bi
-        exs = Set.fromList extras
+  where new = ordNub (extras ++ otherModules bi)
 
 -- | Add extra source dir for generated modules.
 addSrcDir :: BuildInfo -> FilePath -> BuildInfo
 addSrcDir bi extra = bi { hsSourceDirs = new }
-  where new = Set.toList $ old `Set.union` ex
-        old = Set.fromList $ hsSourceDirs bi
-        ex  = Set.fromList [unsafeMakeSymbolicPath extra] -- TODO
```
Comment (fendor) [inline Cabal/src/Distribution/Simple/Build.hs:481]: What happened to the `TODO`?

Comment (fendor) [reply Cabal/src/Distribution/Simple/Build.hs:481]: yeah, neither do I, unfortunately. I hoped you fixed it somehow :D

**Review by ulysses4ever — APPROVED** [review]

File: Cabal/src/Distribution/Simple/Build.hs (hunk 1 above)
Comment (ulysses4ever) [reply Cabal/src/Distribution/Simple/Build.hs:481]: What it meant I have a guess: maybe the author of TODO wished there was no "unsafe"?

**Review by Mikolaj — APPROVED** [review]
LGTM. Given how long it stayed there, let's not worry too much about `unsafeMakeSymbolicPath`.

**Comment by Mikolaj** [comment]
@gbaz: let's merge?

**Pull request merged**